cachi2/core/package_managers/gomod.py

```diff
@@ -827,6 +827,13 @@ def _setup_go_toolchain(go_mod_file: RootedPath) -> Go:
     return go
 
 
+def _disable_telemetry(go: Go, run_params: dict[str, Any]) -> None:
+    telemetry = go(["env", "GOTELEMETRY"], run_params).rstrip()
+    if telemetry and telemetry != "off":
```

Comment:
A nitpick: `telemetry` is already a string, first check is redundant. Also do you really need a check if you are disabling telemetry anyway? I.e.
```suggestion
def _disable_telemetry(go: Go, run_params: dict[str, Any]) -> None:
        log.debug("Disabling Go telemetry")
        go(["telemetry", "off"], run_params)
will apparently do the same.

Reply:
> telemetry is already a string

and if it's empty then that Go toolchain doesn't support telemetry and the command as you suggest will fail, e.g. on Go 1.20 which we can also fallback to (still).

> Also do you really need a check if you are disabling telemetry anyway? I.e.

Yes I do, for the reason above as well as not to spin up another subprocess if I don't need to (i.e. when it's already "off")

Reply:
I see, thank you!